### ISSUE TITLE
`function-whitespace-after` reads a grouping parenthesis as a function's, and either option corrupts a Less or SCSS calculation

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#230]
Found while tidying the comments of the repository: `function-whitespace-after/index.test.js` carried a commented-out case marked "temporarily disable this test until this is fully supported in stylelint". Nothing about it waits on Stylelint — the rule corrupts the code it is pointed at — so the block is removed and the case is written up here.

### What minimal example or steps are needed to reproduce the bug?

A Less calculation carrying a unit behind it:

```less
h1 { width: (@a * 2)px; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/function-whitespace-after": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --custom-syntax postcss-less --config myconfig.json --fix
```

### What did you expect to happen?

Nothing reported. The parenthesis opens no function: it groups an arithmetic expression, and the unit standing behind it belongs to that expression. Less compiles the declaration to a single value.

### What actually happened?

`1:22 Expected whitespace after ")"`, and the fix writes the whitespace in:

```less
h1 { width: (@a * 2) px; }
```

which Less compiles to two values separated by a space instead of one.

The `never` option corrupts the mirror image of the same code: `h1 { width: (@a * 2) px; }`, two values as the author wrote them, is closed up into `h1 { width: (@a * 2)px; }`, an expression carrying a unit. So whichever option a Less project picks, one of the two shapes is rewritten into the other.

`postcss-scss` goes the same way, `h1 { width: (1 + 2)px; }` coming out as `h1 { width: (1 + 2) px; }`. Plain CSS has no grouping parenthesis in a value, so it is untouched: `translate(1px)rotate(2deg)` is fixed to `translate(1px) rotate(2deg)`, which is right.

The case the removed test carried, `((@line-height) * (@lines-to-show))em`, is this same one written with two nested groups.

### What is the cause?

The rule searches the value for every `)` with `styleSearch({ target: ")", functionArguments: "only" })` and hands each one to `checkClosingParen`, which looks at the character standing behind it and at nothing else. Neither asks whether a name stands in front of the parenthesis, so a group is a function as far as this rule is concerned.

`postcss-value-parser` gives a grouping parenthesis a function node with an empty `value`, and `isStandardSyntaxFunction` turns exactly those away — "Function nodes without names are things in parentheses like Sass lists" — but this rule reads no parsed value and calls no such util. `findFunctionArgumentSpans`, which `mediaFeatureColonSpaceChecker` uses, answers the same question over a plain string: a parenthesis opens a call only where an identifier stands in front of it.
